internal/services/serviceprincipals/synchronization_job_resource.go

```diff
@@ -0,0 +1,251 @@
+package serviceprincipals
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/manicminer/hamilton/msgraph"
+	"github.com/manicminer/hamilton/odata"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/helpers"
+	"github.com/hashicorp/terraform-provider-azuread/internal/services/serviceprincipals/parse"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+	"github.com/hashicorp/terraform-provider-azuread/internal/utils"
+	"github.com/hashicorp/terraform-provider-azuread/internal/validate"
+)
+
+func synchronizationJobResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: synchronizationJobResourceCreate,
+		ReadContext:   synchronizationJobResourceRead,
+		UpdateContext: synchronizationJobResourceUpdate,
+		DeleteContext: synchronizationJobResourceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		SchemaVersion: 0,
+
+		Schema: map[string]*schema.Schema{
+			"service_principal_id": {
+				Description:      "The id of the application's service principal for which this synchronization job should be created",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.UUID,
+			},
+			"id": {
+				Description: "Unique synchronization job identifier.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"template_id": {
+				Description: "Identifier of the synchronization template this job is based on.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"state": {
+				Description: "Desired state.",
+				Type:        schema.TypeString,
+				Default:     "Active",
+				Optional:    true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Active",
+					"Disabled",
+				}, false),
+			},
```

Comment:
Can we make this a boolean instead?

```suggestion
			"enabled": {
				Description: "Whether or not the synchronization job is enabled",
				Type:        schema.TypeBool,
				Default:     true,
				Optional:    true,
			},
```

Reply:
I have changed it to use the boolean logic as you suggested.